src/main.rs

```diff
@@ -0,0 +1,149 @@
+use getopts::Options;
+use regex::RegexBuilder;
+use std::{
+    env,
+    fs::{read_to_string, write},
+    io::{stdout, Write},
+    path::Path,
+    process::{self, Command, Stdio},
+};
+
+enum PubKind {
+    Pub,
+    Crate,
+    Super,
+    Private,
+}
+
+fn process(oracle_cmd: &str, p: &Path) -> u64 {
+    let pub_regex = RegexBuilder::new("pub(?:\\s*\\(\\s*(.*?)\\s*\\))?")
+        .multi_line(true)
+        .build()
+        .unwrap();
+    let mut cur_txt = read_to_string(p).unwrap();
+    let mut i = 0;
+    let mut cl = pub_regex.capture_locations();
+    let mut num_changed = 0;
+    while i < cur_txt.len() {
+        print!(".");
+        stdout().flush().ok();
+        let old_txt = cur_txt.clone();
+        let m = match pub_regex.captures_read_at(&mut cl, &old_txt, i) {
+            Some(m) => m,
+            None => break,
+        };
+        let mut kind = if let Some((start, end)) = cl.get(1) {
+            match &cur_txt[start..end] {
+                "crate" => PubKind::Crate,
+                "super" => PubKind::Super,
+                _ => {
+                    // FIXME: this captures things we don't need to deal with (e.g. `pub(self)`),
```

Comment:
A future version of this tool could use a parser (or maybe rust-analyzer) so as to only mutate things that really are visibility modifiers?

Reply:
Probably one should plug into rustc. That's no small job!

Reply:
Yeah, that's why rust-analyzer is probably more attractive. But anyway, we won't do that right now.